src/carbon-price/carbon-price.service.ts

```diff
@@ -63,16 +67,46 @@ export class CarbonPriceService {
         `Processing batch from ${currentStartBlock} to ${batchEndBlock}, found ${tradedEvents.length} events`,
       );
 
+      // Collect quotes for this batch only
+      const historicQuotesToInsert: Partial<HistoricQuote>[] = [];
+      const quoteUpdates: any[] = [];
+
       // Process each traded event
       for (const event of tradedEvents) {
         totalProcessed++;
 
-        const result = await this.processTradeEvent(event, lowercaseTokenMap, deployment);
+        const result = await this.processTradeEvent(
+          event,
+          lowercaseTokenMap,
+          deployment,
+          ethereumRates,
+          deploymentRates,
+          historicQuotesToInsert,
+          quoteUpdates,
+        );
         if (result) {
           totalPricesUpdated++;
         }
       }
 
+      // Batch insert historic quotes for this batch
+      if (historicQuotesToInsert.length > 0) {
+        this.logger.log(
+          `Batch inserting ${historicQuotesToInsert.length} historic quotes for batch ${currentStartBlock}-${batchEndBlock}`,
+        );
+        await this.historicQuoteService.addQuotesBatch(historicQuotesToInsert);
+      }
+
+      // Batch update quote service entries for this batch
+      if (quoteUpdates.length > 0) {
+        this.logger.log(
+          `Processing ${quoteUpdates.length} quote updates for batch ${currentStartBlock}-${batchEndBlock}`,
+        );
+        for (const quoteUpdate of quoteUpdates) {
```

Comment:
This is sequential, are you aiming for it? Opposed to

await Promise.all(
  quoteUpdates.map(q => this.quoteService.addOrUpdateQuote(q))
);